server/service/src/sync/translations/invoice.rs

```diff
@@ -258,6 +259,10 @@ impl SyncTranslation for InvoiceTranslation {
             format!("Unsupported invoice type: {:?}", data._type),
         ))?;
         let mapping = map_legacy(&invoice_type, &data);
+        let default_currency = CurrencyRepository::new(connection)
+            .query_by_filter(CurrencyFilter::new().is_home_currency(true))?
+            .pop()
+            .ok_or(anyhow::Error::msg("Default currency not found"))?;
```

Comment:
Since this is an edge case, this one is every time, but is used "never".

Please use `.unwrap_or_else` or do a:

```rust
let currency_id = match data.currency_id {
  Some(id) => id,
  None => CurrencyRepository::new()...
};
````